Makefile

```diff
@@ -1,23 +1,58 @@
 CONTAINER_NAME := jenkins-server-container
 
 install:
-	./install.sh
+	./scripts/install.sh
 
 clean:
 	docker rm -v $$(docker ps -a -q | grep -v "$$(docker ps -q | xargs | sed 's/ /\\\|/g') ") 2>/dev/null || echo Nothing to do
 	docker rmi $$(docker images --no-trunc | grep none | awk '{print $$3 }') 2>/dev/null || echo Nothing to do
 
 build:
-	./build.sh
+	./scripts/build.sh
 
 start:
-	docker run --name $(CONTAINER_NAME) -p 8080:8080 -v /var/jenkins_home:/var/jenkins_home --restart always -d jenkins_server
+	docker run -d \
+		--name $(CONTAINER_NAME) \
+		-p 8080:8080 \
+		-v /var/jenkins_home:/var/jenkins_home \
+		-v /mnt/backup:/mnt/backup \
+		-v /var/run/docker.sock:/var/run/docker.sock \
+		--restart always \
+		jenkins_server
 
 stop:
-	echo $(CONTAINER_NAME)
+	@echo 'Stopping $(CONTAINER_NAME)'
 	docker stop $(CONTAINER_NAME)
 
 rebuild: build stop clean start
 
-
-.PHONY: install clean build start stop rebuild
+dev:
+	mkdir -p .docker-dev/backup
+	mkdir -p .docker-dev/jenkins_home
+	sudo chown 1000:1000 .docker-dev/jenkins_home .docker-dev/backup
+	docker run -ti --rm \
+		--name jenkins-server-dev \
+		-p 8080:8080 \
+		-v `pwd`/.docker-dev/backup:/mnt/backup \
+		--volumes-from jenkins-server-dind \
+		jenkins_server
+
+dev.dind:
+	docker run -d -ti \
+		--privileged \
+		--name jenkins-server-dind \
+		-v `pwd`/.docker-dev/jenkins_home:/var/jenkins_home \
+		-v `pwd`/.docker-dev/run:/var/run \
+		fgrehm/alpine-dind
```

Comment:
This is a pretty small image based on [Alpine Linux](http://alpinelinux.org/) that is able to run a Docker instance inside it: https://github.com/fgrehm/docker-alpine-dind :smirk: 

The idea is that we can use a dedicated Docker instance "exclusive" to the jenkins server without messing up with our main Docker instance during development.